test/skate_web/channels/detours_channel_test.exs

```diff
@@ -112,13 +91,21 @@ defmodule SkateWeb.DetoursChannelTest do
     test "subscribes to draft detours with initial detours", %{conn: conn, socket: socket} do
       %{id: authenticated_user_id} = SkateWeb.AuthManager.Plug.current_resource(conn)
 
+      :detour |> build() |> with_id(1) |> insert()
+      :detour |> build() |> with_id(2) |> insert()
+      :detour |> build() |> with_id(3) |> insert()
+
+      put(conn, ~p"/api/detours/update_snapshot", %{
+        "snapshot" => :detour_snapshot |> build() |> with_id(4)
+      })
```

Comment:
```suggestion
    test "subscribes to draft detours with initial detours", %{conn: conn, socket: socket, user: user} do
      %{id: authenticated_user_id} = SkateWeb.AuthManager.Plug.current_resource(conn)

      :detour |> build() |> with_id(1) |> insert()
      :detour |> build() |> with_id(2) |> insert()
      :detour |> build() |> with_id(3) |> insert()

      :detour |> build(author: user) |> with_id(4)
      # OR
      :detour |> build(author_id: authenticated_user_id) |> with_id(4)
```

Reply:
Since `author` is set by the factory, I'm not _certain_ that you could use the `author_id` as is, or if you'd need to explicitly set `author` to `nil` in the case of using `author_id` or not.
```
      :detour |> build(author_id: authenticated_user_id, author: nil) |> with_id(4)

Reply:
But since you can get the current user from the test arguments it's probably easier to just use `author: user`